### PR TITLE
Automatically add users who sign up via SSO to organization's group and don't add them based on domain

### PR DESCRIPTION
**Pull request by siggisim**
---

**Version bump**: Patch

### CODE REVIEW

**Comment by bduffany** [comment]
PR number 1000!!! :medal_sports: :partying_face:

File: enterprise/server/auth/auth.go [hunk 1]
```diff
@@ -223,9 +224,10 @@ type oidcAuthenticator struct {
 	slug         string
 }
 
-func extractToken(issuer string, idToken *oidc.IDToken) (*userToken, error) {
+func extractToken(issuer string, slug string, idToken *oidc.IDToken) (*userToken, error) {
```
Comment (tylerwilliams) [inline enterprise/server/auth/auth.go:227]: nit: you can collapse this to:

func extractToken(issuer, slug string, idToken *oidc.IDToken) (*userToken, error) {

File: enterprise/server/backends/userdb/userdb.go [hunk 2]
```diff
@@ -116,10 +116,23 @@ func (d *UserDB) GetGroupByID(ctx context.Context, groupID string) (*tables.Grou
...
+		if err != nil {
+			return err
+		}
+		group = g
+		return nil
+	})
+	return group, err
+}
+
+func (d *UserDB) getGroupByURLIdentifier(ctx context.Context, tx *db.DB, urlIdentifier string) (*tables.Group, error) {
```
Comment (tylerwilliams) [inline enterprise/server/backends/userdb/userdb.go:131]: nice

File: enterprise/server/backends/userdb/userdb.go [hunk 3]
```diff
@@ -518,7 +538,8 @@ func (d *UserDB) createUser(ctx context.Context, tx *db.DB, u *tables.User) erro
 	}
 	emailDomain := emailParts[1]
 
-	if d.env.GetConfigurator().GetAppAddUserToDomainGroup() {
+	// If the user was created via a group - don't add it to a group based on domain.
```
Comment (tylerwilliams) [inline enterprise/server/backends/userdb/userdb.go:541]: I'm having trouble parsing this comment -- can you explain the "created via a group" part?

**Review by tylerwilliams — APPROVED** [review]
LGTM w/ nits

File: enterprise/server/auth/auth.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/auth/auth.go]: Done

File: enterprise/server/backends/userdb/userdb.go (hunk 2 above)
Comment (siggisim) [reply enterprise/server/backends/userdb/userdb.go:131]: Transactions are fun

File: enterprise/server/backends/userdb/userdb.go (hunk 3 above)
Comment (siggisim) [reply enterprise/server/backends/userdb/userdb.go]: Added some more detail

**Pull request merged**